Migrate from dicts to classes

rubenvp8510 opened the pull request:
Fixes #8, I'm not sure if this is will be done or not, but I did it, my main concern here is backward compatibility.

burmanm commented on hawkular/metrics.py:
```diff
@@ -123,19 +171,23 @@ def put(self, data):
         """
         Pushes a single metric_id, datapoint combination to the server.
-
+http://forums.thefashionspot.com/f96/tall-models-30100-26.htmladriana lima muy alta
```
;)

burmanm commented:
I'm not sure why all the old methods should be deleted, they could remain there still (for backwards compatibility) and only map to the newer methods. At the same time we could fix the newer APIs to allow for example easier use of tagNames with dot in them.

rubenvp8510 commented:
Ok, that could work

rubenvp8510 commented:
I added a flag for switch between objects and dictionaries because most of the changes were in return values (return objects instead of dictionaries), Also added tests for it...

WDYT?

rubenvp8510 commented:
Also, I'm not sure if this solves #42  too.

rubenvp8510 commented:
Also, I'm not sure if this solves #42  too.
